linker/Linker.Steps/MarkStep.cs

```diff
@@ -1587,5 +1598,116 @@ protected virtual void MarkInterfaceImplementation (TypeDefinition type, Interfa
 			MarkCustomAttributes (iface);
 			MarkType (iface.InterfaceType);
 		}
+
+
+		void MarkSomethingUsedViaReflection (string reflectionMethod, Action<Collection<Instruction>, string, TypeDefinition, BindingFlags> markMethod, Collection<Instruction> instructions)
+		{
+			for (var i = 0; i < instructions.Count; i++) {
+				var instruction = instructions [i];
+				if (instruction.OpCode == OpCodes.Call || instruction.OpCode == OpCodes.Callvirt) {
+					var methodBeingCalled = instruction.Operand as MethodReference;
+					if (methodBeingCalled != null && methodBeingCalled.DeclaringType.Name == "Type" && methodBeingCalled.DeclaringType.Namespace == "System")
```

Comment:
Please use `continue` instead of deep nesting

Reply:
I'll make this change.